src/utils.cpp

```diff
@@ -42,12 +42,25 @@ std::string sysError() noexcept
     return strerror(errno);
 }
 
-Utils::EVPKeyPtr readPublicKey(const std::string& fileName)
+Utils::EVPKeyPtr readPublicKey(const std::string& src)
 {
-    FilePtr fp(fopen(fileName.c_str(), "rb"));
-    if (!fp)
-        throw JWTXX::Key::Error("Can't open key file '" + fileName + "'. " + sysError());
-    return Utils::EVPKeyPtr(PEM_read_PUBKEY(fp.get(), nullptr, nullptr, nullptr));
+    FilePtr fp(fopen(src.c_str(), "rb"));
+
+    // src is file name
+    if (fp)
+        return Utils::EVPKeyPtr(PEM_read_PUBKEY(fp.get(), nullptr, nullptr, nullptr));
+
+    // src is key data
+    unsigned char *decodedData = new unsigned char [src.size()];
+    EVP_DecodeBlock(decodedData, reinterpret_cast<const unsigned char*>(src.data()), static_cast<int>(src.size()));
```

Comment:
we can use PEM_read_bio_PUBKEY instead of decoding the base64 by ourself.

Reply:
@sivachandran  We have to pass string in PEM format if we use `PEM_read_bio_PUBKEY`. This code allows us to use a string without any formatting.
See discussion here
http://www.archivum.info/openssl-users@openssl.org/2010-05/00092/RE-PEM_read_bio_PUBKEY-does-not-do-the-trick.html

Reply:
@AntonQubecinema I'm recommending we pass with `BEGIN` and `END` marker. It just our use case to keep the public key without these marker. But if you look at the general use case the public key would always carry these markers.

Reply:
@AntonQubecinema As we are going to contribute this functionality it is better to keep the input format generic. In html2dcp we can construct the PEM public key from the modulus and exponent we have.